data-sdk/src/main/java/com/logicmonitor/sdk/data/internal/BatchingCache.java

```diff
@@ -182,19 +183,35 @@ public ApiResponse<String> makeRequest(
 
     final String companyUrl = Configuration.setCompany();
     apiClient.setBasePath(companyUrl);
-    Call call =
-        apiClient.buildCall(
-            apiClient.getBasePath(),
-            path,
-            method,
-            queryParams,
-            collectionQueryParams,
-            body,
-            headersParams,
-            cookieParams,
-            formParams,
-            authSetting,
-            apiCallback);
+    if (method.equalsIgnoreCase("PUT") || method.equalsIgnoreCase("PATCH")) {
+      call =
```

Comment:
Can you please add a comment on why we have this loop..

So I assume this is a common call for ingesting metrics and update properties.. So in case of update operation, we need only the latest data so that's why this condition is.. right?? If yes, please add the comment here

Reply:
We do need this loop as we are using "okhttp3" so body with patch and put is not executed properly, only body.get(0) i.e. object for the same is send while in other scenario list is send as body....
when we are sending same list as a body for PATCH and PUT it gives "bad request" as internally we have serialisation for body (in okhttp3)  for which body is not formed correctly.

Reply:
@Avleen-Kathuria I think @mukundneharkar  is asking for a comment in code.